astroquery/nasa_exoplanet_archive/core.py

```diff
@@ -483,49 +484,34 @@ def _parse_result(self, response, verbose=False):
         # For backwards compatibility, add a `sky_coord` column with the coordinates of the object
         # if possible
         if "ra" in data.columns and "dec" in data.columns:
-            data["sky_coord"] = SkyCoord(
-                ra=u.Quantity(data["ra"], unit=u.deg), dec=u.Quantity(data["dec"], unit=u.deg)
-            )
+            data["sky_coord"] = SkyCoord(ra=data["ra"], dec=data["dec"], unit=u.deg)
 
         if not data:
             warnings.warn("Query returned no results.", NoResultsWarning)
 
         return data
 
-    def _handle_deprecated_arguments(self, **kwargs):
+    def _handle_all_columns_argument(self, **kwargs):
         """
-        Deal with arguments that were exposed by earlier versions of the interface
+        Deal with the ``all_columns`` argument that was exposed by earlier versions
 
-        This method will warn users about these deprecated arguments and update the query syntax
-        to use ``select='*'`` instead of ``all_columns``.
+        This method will warn users about this deprecated argument and update the query syntax
+        to use ``select='*'``.
         """
-        show_progress = kwargs.pop("show_progress", None)
-        if show_progress is not None:
-            warnings.warn(
-                "Argument 'show_progress' has been deprecated, and will have no effect",
-                AstropyDeprecationWarning,
-            )
-
-        table_path = kwargs.pop("table_path", None)
-        if table_path is not None:
-            warnings.warn(
-                "Argument 'table_path' has been deprecated, and will have no effect",
-                AstropyDeprecationWarning,
-            )
-
-        all_columns = kwargs.pop("all_columns", None)
-        if all_columns is not None:
-            warnings.warn(
-                "Argument 'all_columns' has been deprecated in favor of 'select=\"*\"'",
-                AstropyDeprecationWarning,
-            )
+        # We also have to manually pop these arguments from the dict because
+        # `deprecated_renamed_argument` doesn't do that for some reason
```

Comment:
Hmm, this is a bug then, but agree that we cannot do much about it here.

Reply:
@pllim - do you recall anything about it upstream?

Reply:
This popped to mind but not sure if it is what you are experiencing. astropy/astropy#9914

Reply:
Hmm. Anyway, that whole decorator business would be better to be nuked and added either upstream or into a separate package rather than tried to be constantly patched over.

Reply:
hah, astropy 3.1 might have popped these args, thus the CI failure.  Anyway, given that `all_columns` still has a functionality, I remove it from the deprecated arguments list, and call it a day. Given the whole method is deprecated, deprecating the arguments wasn't strictly necessary (though it's good to get rid of the ones not doing anything).

Reply:
Brutal! Thanks for tracking this down!